site/src/pages/ManagementSettingsPage/IdpSyncPage/IdpSyncPageView.tsx

```diff
@@ -19,22 +23,57 @@ import {
 	TableLoaderSkeleton,
 	TableRowSkeleton,
 } from "components/TableLoader/TableLoader";
+import { TabLink, Tabs, TabsList } from "components/Tabs/Tabs";
 import type { FC } from "react";
+import { useSearchParams } from "react-router-dom";
 import { MONOSPACE_FONT_FAMILY } from "theme/constants";
 import { docs } from "utils/docs";
+import { ExportPolicyButton } from "./ExportPolicyButton";
+import { IdpPillList } from "./IdpPillList";
 
-export type IdpSyncPageViewProps = {
-	oidcConfig: OIDCConfig | undefined;
-};
+interface IdpSyncPageViewProps {
+	groupSyncSettings: GroupSyncSettings | undefined;
+	roleSyncSettings: RoleSyncSettings | undefined;
+	groups: Group[] | undefined;
+	organization: Organization;
+}
+
+export const IdpSyncPageView: FC<IdpSyncPageViewProps> = ({
+	groupSyncSettings,
+	roleSyncSettings,
+	groups,
+	organization,
+}) => {
+	const [searchParams] = useSearchParams();
+	const groupsMap = new Map<string, string>();
+	if (groups) {
+		for (const group of groups) {
+			groupsMap.set(group.id, group.display_name || group.name);
+		}
+	}
+
+	const getGroupNames = (groupIds: readonly string[]) => {
+		return groupIds.map((groupId) => groupsMap.get(groupId) || groupId);
+	};
+
+	const tab = searchParams.get("tab") || "groups";
+
+	const groupMappingCount = groupSyncSettings?.mapping
+		? Object.entries(groupSyncSettings.mapping).length
+		: 0;
+	const roleMappingCount = roleSyncSettings?.mapping
+		? Object.entries(roleSyncSettings.mapping).length
+		: 0;
+
+	const rolePolicy =
+		roleSyncSettings?.field && roleSyncSettings.mapping
+			? JSON.stringify(roleSyncSettings, null, 2)
+			: null;
+	const groupPolicy =
+		groupSyncSettings?.field && groupSyncSettings.mapping
+			? JSON.stringify(groupSyncSettings, null, 2)
+			: null;
```

Comment:
This is dangerous – we need to be super, super careful with any `JSON.stringify` calls inside components:
1. Stringifying on every render can be slow
2. `JSON.stringify` can throw, so if it fails, we might blow up the entire app